rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromReturnDirectArrayRector/Fixture/parent_and_child_return_array.php.inc

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnDirectArrayRector\Fixture;
+
+/**
+ * Both classes have to be in the same fixture, as the one that needs to be skipped is the parent class
+ */
+class ParentAndChildReturnArray
+{
+    public function get()
+    {
+        return [];
+    }
+}
+
+class ChildReturnArray extends ParentAndChildReturnArray
+{
+    public function get()
+    {
+        return [1];
+    }
+
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnDirectArrayRector\Fixture;
+
+/**
+ * Both classes have to be in the same fixture, as the one that needs to be skipped is the parent class
+ */
+class ParentAndChildReturnArray
+{
+    public function get()
+    {
+        return [];
+    }
+}
+
+class ChildReturnArray extends ParentAndChildReturnArray
+{
+    public function get(): array
+    {
+        return [1];
+    }
```

Comment:
Add return type to only child is expected.

To change parent class method, the collector or special rule may be needed to collect all child methods that must have return type, and the return type must be equal or sub type.